include/ua_plugin_history_data_service.h

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ *    Copyright 2018 (c) basysKom GmbH <opensource@basyskom.com> (Author: Peter Rustler)
+ */
+
+#ifndef UA_PLUGIN_HISTORY_DATA_SERVICE_H_
+#define UA_PLUGIN_HISTORY_DATA_SERVICE_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "ua_types.h"
+#include "ua_server.h"
+
+struct UA_HistoryDataService;
+typedef struct UA_HistoryDataService UA_HistoryDataService;
+
+struct UA_HistoryDataService {
+    void *context;
+
+    void
+    (*deleteMembers)(UA_HistoryDataService *service);
+
+    /* This function will be called when a node value is set.
+     * Use this to insert data into your database(s) if polling is not suitable
+     * and you need to get all data changes.
+     * Set it to NULL if you do not need it.
+     *
+     * context is the context of the UA_HistoryDataService. UA_HistoryDataService.context
+     * server is the server this node lives in.
+     * sessionId and sessionContext identify the session which set this value.
+     * nodeId the node id for which data was set.
+     * value the new value.
+     */
+    void
+    (*setValue)(void *context,
```

Comment:
The argument order should be
`UA_Server *server, void *haContext, const UA_NodeId *sessionId, void *sessionContext, const UA_NodeId *nodeId, void nodeContext, const UA_DataValue *value`